util/argo/argo.go

```diff
@@ -506,6 +506,8 @@ func getDestinationServer(ctx context.Context, db db.ArgoDB, clusterName string)
 		return "", fmt.Errorf("there are %d clusters with the same name: %v", len(servers), servers)
 	} else if len(servers) == 0 {
 		return "", fmt.Errorf("there are no clusters with this name: %s", clusterName)
+	} else if servers[0] == "" {
```

Comment:
The `server[0]` is empty only if the Argo CD settings have an invalid cluster configuration ( cluster with missing server field  ). I think instead of trying to handle it here we need to find how such cluster configuration was created. 

I found another possible reason of #4852 : https://github.com/argoproj/argo-cd/issues/4852#issuecomment-730509010

Reply:
@alexmt You are right, if --validate=false then we won't call 
https://github.com/argoproj/argo-cd/blob/59443ad18585258ba8c49d62b3a5bf5ad51a656a/util/argo/argo.go#L294